File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/paintthenurikabe/data/PaintTheNurikabeDocument.java
package com.zwstudio.logicpuzzlesandroid.puzzles.paintthenurikabe.data;

import com.zwstudio.logicpuzzlesandroid.common.data.GameDocument;
import com.zwstudio.logicpuzzlesandroid.common.data.MoveProgress;
import com.zwstudio.logicpuzzlesandroid.common.domain.Position;
import com.zwstudio.logicpuzzlesandroid.puzzles.paintthenurikabe.domain.PaintTheNurikabeGame;
import com.zwstudio.logicpuzzlesandroid.puzzles.paintthenurikabe.domain.PaintTheNurikabeGameMove;
import com.zwstudio.logicpuzzlesandroid.puzzles.paintthenurikabe.domain.PaintTheNurikabeObject;

import org.androidannotations.annotations.EBean;

@EBean
public class PaintTheNurikabeDocument extends GameDocument<PaintTheNurikabeGame, PaintTheNurikabeGameMove> {
    protected void saveMove(PaintTheNurikabeGameMove move, MoveProgress rec) {
        rec.row = move.p.row;
        rec.col = move.p.col;
        rec.intValue1 = move.obj.ordinal();
    }
    public PaintTheNurikabeGameMove loadMove(MoveProgress rec) {
        return new PaintTheNurikabeGameMove() {{
            p = new Position(rec.row, rec.col);
            obj = PaintTheNurikabeObject.values()[rec.intValue1];
        }};
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/paintthenurikabe/PaintTheNurikabeDocument.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.paintthenurikabe

import android.content.Context
import com.zwstudio.logicpuzzlesandroid.common.data.GameDocument
import com.zwstudio.logicpuzzlesandroid.common.data.MoveProgress
import com.zwstudio.logicpuzzlesandroid.common.domain.Position

class PaintTheNurikabeDocument(context: Context) : GameDocument<PaintTheNurikabeGameMove>(context) {
    override fun saveMove(move: PaintTheNurikabeGameMove, rec: MoveProgress) {
        rec.row = move.p.row
        rec.col = move.p.col
        rec.intValue1 = move.obj.ordinal
    }

    override fun loadMove(rec: MoveProgress) =
        PaintTheNurikabeGameMove(Position(rec.row, rec.col), PaintTheNurikabeObject.values()[rec.intValue1])
}